src/Transports/MassTransit.Azure.ServiceBus.Core/AzureServiceBusTransport/MessageSessionSagaRepository.cs

```diff
@@ -12,7 +15,24 @@ public static ISagaRepository<T> Create<T>()
 
             var repositoryFactory = new MessageSessionSagaRepositoryContextFactory<T>(consumeContextFactory);
 
-            return new SagaRepository<T>(repositoryFactory);
+            return new SagaRepository<T>(repositoryFactory, new TempQuerySagaRepositoryContextFactory<T>());
+        }
+
+
+        class TempQuerySagaRepositoryContextFactory<TSaga> :
```

Comment:
Seems like this could be a built-in type if no query repository is provided in the constructor?

Reply:
I didn't get that

Reply:
Aren't there other saga repositories that do not support queries? Why is this one unique?

Reply:
from what I see that's the only one

Reply:
Nor does DynamodDb.

Reply:
hm, but it supports `Load(id)`